docs/guide/extensibility/plugins/input_handlers.md

```diff
@@ -0,0 +1,504 @@
+can be placed in any package folder,
+including "User".
+
+`simple_plugin.py`:
+``` py
+import sublime_plugin
+
+
+class SimpleCommand(sublime_plugin.TextCommand):
+    def run(self, edit, text):
+        for region in self.view.sel():
+            self.view.replace(edit, region, text)
+```
+
+`Simple.sublime-commands`:
+``` json
+[
+    { "caption": "Simple Command", "command": "simple" },
+]
+```
+
+
+### The `*Command.input` Method
+
+When implementing a command,
+it receives keyed arguments to its `run` method.
+When a parameter in the signature provides no default value,
+it can only be called if arguments are provided for all such parameters.
+Calling a command with too few arguments will fail
+and cause an exception to be printed to the console.
+
+```
+>>> window.run_command("simple")
+Traceback (most recent call last):
+  File "/opt/sublime_text/Lib/python38/sublime_plugin.py", line 1270, in run_
+    return self.run()
+TypeError: run() missing 1 required positional argument: 'text'
+```
+
+In a situation like this,
+a command may implement [the `input` method][api-TextCommand]
+and return an input handler instance
+that provides Sublime Text
+with the necessary information 
+to display an input handler.
+
+``` py {9-10}
+import sublime_plugin
+
+
+class SimpleCommand(sublime_plugin.TextCommand):
+    def run(self, edit, text):
+        for region in self.view.sel():
+            self.view.replace(edit, region, text)
+
+    def input(self, args):
+        return MyTextInputHandler()
+```
+
+The `input` function takes an `args` parameter
+that is a dict of all currently known arguments to the command.
+Since we know that our only required argument (`text`)
+is missing at this point,
+we won't use the parameter.
+
+We haven't defined `MyTextInputHandler` yet,
+so let's do that.
+
+
+### Subclassing `TextInputHandler`
+
+To create a simple input handler for text,
+we create a subclass of [`sublime_plugin.TextInputHandler`][api-TextInputHandler].
+In our subclass,
+we can override specific methods.
+For the most basic functionality,
+we need `name`.
+Additionally, for convenience,
+we define `placeholder`.
+
+``` py
+class MyTextInputHandler(sublime_plugin.TextInputHandler):
+    def name(self):
+        return "text"
+
+    def placeholder(self):
+        return "Text to insert"
+```
+
+And that's it.
+Here is what it looks like:
+
+<video controls src="./images/simple_input_handler.mp4" />
+
+::: tip
+Of course, you can still call the command like before
+from a key binding or via the console.
+When all required arguments are provided,
+the input handler will be skipped
+and the command run immediately.
+:::
+
+
+### Rendering a Preview
+
+The `preview` method is called
+for every modification of the entered text
+and allows to show a small preview
+below the Command Palette.
+The preview can either be pure text
+or can use [minihtml][]
+for a markup-enabled format.
+
+[minihtml]: https://www.sublimetext.com/docs/3/minihtml.html
```

Comment:
target new docs

Reply:
One comment would have been enough.

Alas, no, I'm not referencing the new docs until they have been made public. I will, however, update all urls as soon as they are public, which will be a simple `sed` call.